Lib/socketserver.py

```diff
@@ -639,6 +639,7 @@ class ThreadingMixIn:
     # For non-daemonic threads, list of threading.Threading objects
     # used by server_close() to wait for all threads completion.
     _threads = None
+    _threads_lock = threading.Lock()
```

Comment:
Wait. Written like that, *all* class instances share the same lock. It seems wrong to me. Each instance must have its own lock, no?

Reply:
Yes, I was careless. You are totally right.
Thank you for your comment. I fixed this.